src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBeekeeper.java

```diff
@@ -0,0 +1,259 @@
+
+import com.ldtteam.blockout.views.Window;
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.IColonyView;
+import com.minecolonies.api.colony.buildings.ModBuildings;
+import com.minecolonies.api.colony.buildings.registry.BuildingEntry;
+import com.minecolonies.api.colony.jobs.IJob;
+import com.minecolonies.api.entity.citizen.Skill;
+import com.minecolonies.api.util.NBTUtils;
+import com.minecolonies.api.util.constant.NbtTagConstants;
+import com.minecolonies.coremod.Network;
+import com.minecolonies.coremod.client.gui.WindowHutBeekeeper;
+import com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker;
+import com.minecolonies.coremod.colony.jobs.JobBeekeeper;
+import com.minecolonies.coremod.network.messages.server.colony.building.beekeeper.BeekeeperSetHarvestHoneycombsMessage;
+import net.minecraft.item.Items;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.Tuple;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.minecolonies.api.util.constant.BuildingConstants.CONST_DEFAULT_MAX_BUILDING_LEVEL;
+
+public class BuildingBeekeeper extends AbstractBuildingWorker
+{
+    /**
+     * Description of the job executed in the hut.
+     */
+    private static final String BEEKEEPER = "beekeeper";
+
+    /**
+     * List of hives.
+     */
+    private Set<BlockPos> hives = new HashSet<>();
+
+    /**
+     * Wether the beekeeper should harvest honeycombs or honey bottles
+     */
+    private boolean harvestHoneycombs = true;
+
+    /**
+     * The abstract constructor of the building.
+     *
+     * @param c the colony
+     * @param l the position
+     */
+    public BuildingBeekeeper(@NotNull final IColony c, final BlockPos l)
+    {
+        super(c, l);
+        keepX.put(stack -> Items.SHEARS == stack.getItem(), new Tuple<>(1, true));
+        keepX.put(stack -> Items.GLASS_BOTTLE == stack.getItem(), new Tuple<>(4, true));
+    }
+
+    /**
+     * The abstract method which creates a job for the building.
+     *
+     * @param citizen the citizen to take the job.
+     * @return the Job.
+     */
+    @NotNull
+    @Override
+    public IJob createJob(final ICitizenData citizen)
+    {
+        return new JobBeekeeper(citizen);
+    }
+
+    /**
+     * The abstract method which returns the name of the job.
+     *
+     * @return the job name.
+     */
+    @NotNull
+    @Override
+    public String getJobName()
+    {
+        return BEEKEEPER;
+    }
+
+    /**
+     * Primary skill getter.
+     *
+     * @return the primary skill.
+     */
+    @NotNull
+    @Override
+    public Skill getPrimarySkill()
+    {
+        return Skill.Stamina; //TODO
```

Comment:
Dexterity -> primary
Adaptability -> Secondary

Were planned for this worker

Reply:
ok (that's why I put the todo there)